main.go

```diff
@@ -1,97 +1,121 @@
 package main
 
 import (
+	"database/sql"
 	"encoding/json"
 	"fmt"
 	"log"
 	"net/http"
 	"strconv"
-)
+	"strings"
 
-/*
-example:
-	*{"name": "Ilya", "hobby": "Programming"}
-	*localhost:9000/?id=1
-*/
+	_ "github.com/go-sql-driver/mysql"
+)
 
 // User represents user with name and hobby strings fields.
 type User struct {
-	Name  string `json:"name"`
-	Hobby string `json:"hobby"`
+	ID      int64  `json:"id"`
+	Name    string `json:"name"`
+	Balance uint   `json:"balance"`
 }
 
-var idUserDB = make(map[int]User)
-
-var nameHobbyDB = make(map[string]string)
+// Server represents server with pointer to a DB.
+type Server struct {
+	DB *sql.DB
+}
 
-var nextID int
+func dbConn() (db *sql.DB, err error) {
+	dbDriver := "mysql"
+	dbUser := "root"
+	dbPass := "Sql_DB_tournament1"
+	dbName := "tournament_db"
 
-func main() {
-	http.HandleFunc("/", handler)
-	err := http.ListenAndServe("localhost:9000", nil)
+	db, err = sql.Open(dbDriver, fmt.Sprintf("%s:%s@/%s", dbUser, dbPass, dbName))
 	if err != nil {
-		log.Fatal("failed to start server", err)
+		return db, err
 	}
+	return db, nil
 }
 
-func handler(writer http.ResponseWriter, req *http.Request) {
-	switch req.Method {
-	case http.MethodGet:
-		getUser(writer, req)
-	case http.MethodPost:
-		addUser(writer, req)
+func main() {
+	db, err := dbConn()
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer db.Close()
+	s := Server{
+		DB: db,
+	}
+	http.HandleFunc("/user", s.addUser)
+	http.HandleFunc("/user/", s.getUser)
+	err = http.ListenAndServe("localhost:9000", nil)
+	if err != nil {
+		log.Fatal(err)
 	}
 }
 
-func addUser(writer http.ResponseWriter, req *http.Request) {
+func (s *Server) addUser(writer http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodPost {
+		http.NotFound(writer, req)
+		return
+	}
 	var user User
 	err := json.NewDecoder(req.Body).Decode(&user)
 	if err != nil {
 		writer.WriteHeader(http.StatusBadRequest)
 		fmt.Fprintf(writer, "cannot decode json: %s", err)
 		return
 	}
-	if _, found := nameHobbyDB[user.Name]; found {
-		log.Printf("this user %s with hobby %s already exists", user.Name, user.Hobby)
-		writer.WriteHeader(http.StatusPreconditionFailed)
-		fmt.Fprintf(writer, "this user already exists")
+	insert, err := s.DB.Exec("INSERT INTO user(name,balance) VALUES(?,?)", user.Name, 0)
+	if err != nil {
+		writer.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(writer, "insertion has failed: %s", err)
```

Comment:
I think `could not add user` may be better,  b/c there can be lots of insertions in your code